server-adapters/resteasy-undertow-spring/src/test/java/org/jboss/resteasy/springmvc/test/spring/AsynchSpringTest.java

```diff
@@ -0,0 +1,209 @@
+      response.close();
+
+      builder = client.target(jobUrl1).request();
+      response = builder.get();
+      Assert.assertEquals(HttpServletResponse.SC_GONE, response.getStatus());
+      response.close();
+
+      // test its still there
+      Thread.sleep(1000);
+      builder = client.target(jobUrl2).request();
+      response = builder.get();
+      Assert.assertEquals(HttpServletResponse.SC_OK, response.getStatus());
+      Assert.assertEquals("content", response.readEntity(String.class));
+
+      // delete and test delete
+      response = builder.delete();
+      Assert.assertEquals(HttpServletResponse.SC_NO_CONTENT, response.getStatus());
+
+      response = builder.get();
+      Assert.assertEquals(HttpServletResponse.SC_GONE, response.getStatus());
+      response.close();
+   }
+
+   private AsynchronousDispatcher getDispatcher(UndertowJaxrsSpringServer server) throws Exception {
+      DispatcherServlet servlet = (DispatcherServlet) server.getManager().getDeployment().getServlets().getManagedServlet(UndertowJaxrsSpringServer.SERVLET_NAME).getServlet().getInstance();
+      AsynchronousDispatcher dispatcher = (AsynchronousDispatcher) servlet.getWebApplicationContext().getBean("resteasy.dispatcher");
+      return dispatcher;
+   }
+
+   @Test
+   public void testasync3() throws Exception {
+      Client client = ResteasyClientBuilder.newClient();
+      Response response = null;
+
+      // test readAndRemove
+
+      getDispatcher(server).setMaxCacheSize(10);
+      latch = new CountDownLatch(1);
+      Builder builder = client.target("http://localhost:" + TestPortProvider.getPort() + "?asynch=true").request();
+      response = builder.post(Entity.entity("content", "text/plain"));
+      Assert.assertEquals(HttpServletResponse.SC_ACCEPTED, response.getStatus());
+      String jobUrl2 = response.getHeaderString(HttpHeaders.LOCATION);
+      Assert.assertTrue(latch.await(3, TimeUnit.SECONDS));
+      response.close();
+
+      Thread.sleep(1000);
+      // test its still there
+      builder = client.target(jobUrl2).request();
+      response = builder.post(Entity.entity("content", "text/plain"));
+      Assert.assertEquals(HttpServletResponse.SC_OK, response.getStatus());
+      Assert.assertEquals("content", response.readEntity(String.class));
+
+      builder = client.target(jobUrl2).request();
+      response = builder.get();
+      Assert.assertEquals(HttpServletResponse.SC_GONE, response.getStatus());
+      response.close();
+
+   }
+
+   @Path("/")
+   public static class MyResource {
+      @POST
+      public String post(String content) throws Exception {
+         Thread.sleep(1500);
+         latch.countDown();
+
+         return content;
+      }
+
+      @PUT
+      public void put(String content) throws Exception {
+//         System.out.println("IN PUT!!!!");
+         Assert.assertEquals("content", content);
+         Thread.sleep(500);
+//         System.out.println("******* countdown ****");
```

Comment:
Can we get rid of the comments here and few lines above while fixing the other stuff?

Reply:
okay!